tests/Aql/AqlValueMemoryLayoutTest.cpp

```diff
@@ -0,0 +1,419 @@
+    0xff, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(4294967295ULL), expected);
+}
+
+TEST(AqlValueMemoryLayoutTest, UnsignedLargerValues48Bit_4294967296) {
+  // 01 00 00 00 00
+  uint8_t const expected[] = { 
+    AqlValue::AqlValueType::VPACK_INLINE_INT48, 0x2c, 0x00, 0x00, 0x00, 0x00, 0x01, UNINITIALIZED,
+    0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(4294967296ULL), expected);
+}
+
+TEST(AqlValueMemoryLayoutTest, UnsignedLargerValues48Bit_549755813887) {
+  // 7f ff ff ff ff
+  uint8_t const expected[] = { 
+    AqlValue::AqlValueType::VPACK_INLINE_INT48, 0x2c, 0xff, 0xff, 0xff, 0xff, 0x7f, UNINITIALIZED,
+    0xff, 0xff, 0xff, 0xff, 0x7f, 0x00, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(549755813887ULL), expected);
+}
+
+TEST(AqlValueMemoryLayoutTest, UnsignedLargerValues48Bit_549755813888) {
+  // 7f ff ff ff ff
+  uint8_t const expected[] = { 
+    AqlValue::AqlValueType::VPACK_INLINE_INT48, 0x2d, 0x00, 0x00, 0x00, 0x00, 0x80, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x80, 0x00, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(549755813888ULL), expected);
+}
+
+TEST(AqlValueMemoryLayoutTest, UnsignedLargerValues48Bit_1099511627776) {
+  // 01 00 00 00 00 00
+  uint8_t const expected[] = { 
+    AqlValue::AqlValueType::VPACK_INLINE_INT48, 0x2d, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(1099511627776ULL), expected);
+}
+
+TEST(AqlValueMemoryLayoutTest, UnsignedLargerValues48Bit_140737488355327) {
+  // 7f ff ff ff ff ff
+  uint8_t const expected[] = { 
+    AqlValue::AqlValueType::VPACK_INLINE_INT48, 0x2d, 0xff, 0xff, 0xff, 0xff, 0xff, 0x7f,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0x7f, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(140737488355327ULL), expected);
+}
+
+// note: in all following tests, the value UNINITIALIZED (0xa5) has a special meaning.
+// if it used, it means that we don't compare the byte, because it is not initialized
+// in certain AqlValue configurations.
+// if you add tests, make sure that none of the test data includes 0xa5!
+TEST(AqlValueMemoryLayoutTest, UnsignedLargerValues64Bit_281474976710654) {
+  // 80 ff ff ff ff ff
+  uint8_t const expected[] = { 
+    AqlValue::AqlValueType::VPACK_INLINE_UINT64, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, 0x2f,
+    0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(281474976710654ULL), expected);
+}
+
+TEST(AqlValueMemoryLayoutTest, UnsignedLargerValues64Bit_281474976710655) {
+  // ff ff ff ff ff ff
+  uint8_t const expected[] = { 
+    AqlValue::AqlValueType::VPACK_INLINE_UINT64, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, UNINITIALIZED, 0x2f,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00
+  };
+  runChecksForUInt64(uint64_t(281474976710655ULL), expected);
```

Comment:
Why is this value encoded as a 64bit value? It only requires 48bits...

Reply:
As discussed, this is necessary because the value is encoded as a signed integer right now.